x/stake/keeper/distribute_sidechain.go

```diff
@@ -264,8 +282,28 @@ func (k Keeper) distributeSingleBatch(ctx sdk.Context, sideChainId string) {
 			distAddrBalanceMap[distAddr.String()] = reward.Amount
 		}
 
-		if _, _, err := k.bankKeeper.AddCoins(ctx, reward.AccAddr, sdk.Coins{sdk.NewCoin(bondDenom, reward.Amount)}); err != nil {
-			panic(err)
+		if reward.CrossStake && sdk.IsUpgrade(sdk.BEP153) {
+			rewardCAoB := types.GetStakeCAoB(reward.AccAddr.Bytes(), types.RewardCAoBSalt)
+			if _, _, err := k.BankKeeper.AddCoins(ctx, rewardCAoB, sdk.Coins{sdk.NewCoin(bondDenom, reward.Amount)}); err != nil {
+				panic(err)
+			}
+			balance := k.BankKeeper.GetCoins(ctx, rewardCAoB).AmountOf(bondDenom)
+			if balance >= types.MinRewardThreshold {
+				if _, err := k.BankKeeper.SendCoins(ctx, rewardCAoB, sdk.PegAccount, sdk.Coins{sdk.NewCoin(bondDenom, balance)}); err != nil {
+					panic(err)
+				}
+				event, err := crossDistributeReward(k, ctx, rewardCAoB, reward.AccAddr, balance)
```

Comment:
Let us do `crossDistributeReward` after the calculation is done.

Reply:
Done